### PR TITLE
Fix msearch_template Response

### PR DESCRIPTION
**Pull request by philkra**
as titled.

### CODE REVIEW

**Review by delvedor — COMMENTED** [review]
@swallez can you take a look at this one? You added this change in https://github.com/elastic/elasticsearch-specification/pull/1034.

File: specification/_global/msearch/types.ts [hunk 1]
```diff
@@ -65,9 +65,5 @@ export class MultiSearchResult<TDocument> {
 
 /** @codegen_names result, failure */
 export type ResponseItem<TDocument> =
-  | MultiSearchItem<TDocument>
+  | SearchResponse<TDocument>
```
Comment (swallez) [inline specification/_global/msearch/types.ts:68]: multisearch response items have a `status` field that indicates the result of each search request. This is present in the [ES source code](https://github.com/elastic/elasticsearch/blob/8754c17a2d6218bce2041cddc572d2a58cef895f/server/src/main/java/org/elasticsearch/action/search/MultiSearchResponse.java#L164) and [the docs](https://www.elastic.co/guide/en/elasticsearch/reference/current/search-multi-search.html#search-multi-search-api-response-body).

Comment (swallez) [inline specification/_global/msearch/types.ts:68]: We need `MultiSearchItem` as multisearch response items have a `status` field that indicates the result of each search request. This is present in the [ES source code](https://github.com/elastic/elasticsearch/blob/8754c17a2d6218bce2041cddc572d2a58cef895f/server/src/main/java/org/elasticsearch/action/search/MultiSearchResponse.java#L164) and explained in [the docs](https://www.elastic.co/guide/en/elasticsearch/reference/current/search-multi-search.html#search-multi-search-api-response-body).

File: specification/_types/Base.ts [hunk 2]
```diff
@@ -73,7 +73,7 @@ export class ErrorResponseBase {
   // If the error is a string, it means that it was not caused by an exception on ES side, but on the HTTP routing layer.
   // This should never happen in clients, because we assume we will never send malformed request.
   error: ErrorCause
-  status: integer
+  status?: integer
```
Comment (swallez) [inline specification/_types/Base.ts:76]: Status is always present in error responses, including in multisearch responses. See [ES code](https://github.com/elastic/elasticsearch/blob/8754c17a2d6218bce2041cddc572d2a58cef895f/server/src/main/java/org/elasticsearch/action/search/MultiSearchResponse.java#L161) and [the docs](https://www.elastic.co/guide/en/elasticsearch/reference/current/search-multi-search.html#search-multi-search-api-response-body) "an object with error message and corresponding status code will be returned in place of the actual search response".

Keeping `error` and `status` required also plays an important role (in the Java client) to disambiguate some non-200 results where ES will return either a valid response or an error depending on the parameters (e.g. delete document that returns 404+error if the index doesn't exist but 404+valid response if the index exist but not the document).

**Review by swallez — CHANGES_REQUESTED** [review]
Looking at the reference material (code + docs), this change seems wrong. What are the recordings that led to it?

**Comment by philkra** [comment]
Closing this in favour of https://github.com/elastic/elasticsearch-specification/pull/1320

**Pull request closed without merging**